### ISSUE TITLE
Errors on files with tabs

### ISSUE THREAD

**Issue by davidgilbertson** [sasstools/sass-lint#62]
I get this error when running the tool:
```
C:\www\mysite\node_modules\sass-lint\lib\rules\indentation.js:51
                  'line': node.content[i + 1].start.line,
                                             ^
TypeError: Cannot read property 'start' of undefined
    at processNode (C:\www\mysite\node_modules\sass-lint\lib\rules\indentation.js:51:46)
    at processNode (C:\www\mysite\node_modules\sass-lint\lib\rules\indentation.js:76:9)
    at processNode (C:\www\mysite\node_modules\sass-lint\lib\rules\indentation.js:76:9)
    at processNode (C:\www\mysite\node_modules\sass-lint\lib\rules\indentation.js:76:9)
    at processNode (C:\www\mysite\node_modules\sass-lint\lib\rules\indentation.js:76:9)
    at Object.module.exports.detect (C:\www\mysite\node_modules\sass-lint\lib\rules\indentation.js:80:5)
    at C:\www\mysite\node_modules\sass-lint\index.js:30:25
    at Array.forEach (native)
    at Function.sassLint.lintText (C:\www\mysite\node_modules\sass-lint\index.js:29:9)
    at C:\www\mysite\node_modules\sass-lint\index.js:73:21
```
I have narrowed it down to a file that has tabs in it, if I replace the tabs with spaces (which I want to do anyway) there is no error.

sass-lint version: 1.1.0
OS: windows 10

**Comment by Snugug** [comment]
Yup, this is a known issue with our AST. In the next release of it the space issue should be resolved. Unfortunately, there's nothing I can do about it at the moment :cry: 

I'm going to keep this open until it's resolved.

**Comment by davidgilbertson** [comment]
Cool, thanks for the quick reply.

**Comment by mven** [comment]
@davidgilbertson I know it's a pita, but you could try to open each file into Sublime Text and convert tabs into spaces or vice versa and it should fix the spacing issue.

**Comment by davidgilbertson** [comment]
@mven Yep already done and that fixed the problem (WebStorm has a nice tabs-to-spaces option).

**Comment by cjolif** [comment]
@Snugug  do you have a link to the original issue?

**Comment by Snugug** [comment]
https://twitter.com/tonyganch/status/637355652835340288

https://twitter.com/tonyganch/status/637355801917657088

**Comment by natwebb** [comment]
I'm getting an identical error from a file that doesn't include any tabs. Here's the scss file in question in its entirety:

    ///////////////////////////////////////////
    ///  Advertiser Dashboard View
    ///////////////////////////////////////////
    
    .analytics-empty-list {
        text-align: center;
     }
    .analytics-redeem {
        color:brown;
     }
    .analytics-capture {
        color:darkorange;
     }
    .analytics-impression {
        color:olivedrab;
    }
    .analytics-refresh {
        font-size: px-to-rem(17);
    }

If I comment out the entire file, I get about eight warnings (most of them innacurate), ending with a final-newline warning, then the same error as above (possibly from trying to read a new file?). If I add the requested final-newline, I get the same error as above from my original file.

There are no tabs, and barely any content. Any uncommented line seems to cause this error. Thoughts?

**Comment by DanPurdy** [comment]
hey @natwebb could you provide a few details to help us track this down,
* sass-lint version
* your config file
* whether this is running from the CLI or gulp/grunt etc

For me copying that example you pasted above and running against 1.2.0 I get the following errors with the default config

```bash
   7:2  warning  Space expected between blocks                                                     empty-line-between-blocks
   9:8  warning  Space expected after `:`                                                          space-after-colon
   9:9  warning  Color 'brown' should be written in its hexadecimal form #a52a2a                   no-color-keywords
   9:9  warning  color literals such as 'brown' should only be used in variable declarations       no-color-literals
  10:2  warning  Space expected between blocks                                                     empty-line-between-blocks
  12:8  warning  Space expected after `:`                                                          space-after-colon
  12:9  warning  Color 'darkorange' should be written in its hexadecimal form #ff8c00              no-color-keywords
  12:9  warning  color literals such as 'darkorange' should only be used in variable declarations  no-color-literals
  13:2  warning  Space expected between blocks                                                     empty-line-between-blocks
  15:8  warning  Space expected after `:`                                                          space-after-colon
  15:9  warning  color literals such as 'olivedrab' should only be used in variable declarations   no-color-literals
  15:9  warning  Color 'olivedrab' should be written in its hexadecimal form #6b8e23               no-color-keywords
  16:2  warning  Space expected between blocks                                                     empty-line-between-blocks
```

**Comment by davidgilbertson** [comment]
@natwebb I was having the same issue, I deleted every sass file but one, removed all rules, definitely no tabs, and still the error.

I wound up just excluding the `indentation` rule in the config file.

@DanPurdy This is version `1.1.0` on Windows 10 4bit

Exact error:
```
TypeError: Cannot read property 'start' of undefined
    at processNode (g:\web\davidg-site\node_modules\sass-lint\lib\rules\indentation.js:53:46)
    at processNode (g:\web\davidg-site\node_modules\sass-lint\lib\rules\indentation.js:78:9)
    at processNode (g:\web\davidg-site\node_modules\sass-lint\lib\rules\indentation.js:78:9)
    at Object.module.exports.detect (g:\web\davidg-site\node_modules\sass-lint\lib\rules\indentation.js:82:5)
    at g:\web\davidg-site\node_modules\sass-lint\index.js:30:25
    at Array.forEach (native)
    at Function.sassLint.lintText (g:\web\davidg-site\node_modules\sass-lint\index.js:29:9)
    at g:\web\davidg-site\node_modules\sass-lint\index.js:73:21
    at Array.forEach (native)
    at Function.sassLint.lintFiles (g:\web\davidg-site\node_modules\sass-lint\index.js:72:9)
```

Config file:
```
files:
  include: 'app/**/*.scss'
  ignore: ['node_modules', 'app/components/Gallery/photoswipe.*']
rules:
  # Extends
  extends-before-mixins: 1
  extends-before-declarations: 1
  placeholder-in-extend: 1

  # Mixins
  mixins-before-declarations: 1

  # Line Spacing
  one-declaration-per-line: 1
  empty-line-between-blocks: 1
  single-line-per-selector: 1

  # Disallows
  no-debug: 1
  no-empty-rulesets: 1
  no-extends: 0
  no-ids: 1
  no-important: 1
  no-warn: 1

  # Style Guide
  # indentation has a bug in v1.2 so not testing for it
  # https://github.com/sasstools/sass-lint/issues/62
  indentation: 1
  leading-zero: 0
  nesting-depth: 1
  property-sort-order: 0
  quotes: 1

  # Inner Spacing
  space-after-comma: 1
  space-before-colon: 1
  space-after-colon: 1
  space-before-brace: 1
  space-before-bang: 1
  space-after-bang: 1
  space-between-parens: 1

  # Final Items
  trailing-semicolon: 1
  final-newline: 1
```
(note that I've set indentation to 1 to replicate the error, but I've got it set to 0 normally so I can use the tool)

I'm running this via the cli with the npm script `sass-lint --no-exit --verbose`.

I can't install `1.2.0` because I'm getting this when I try: `shasum check failed for C:\Users\David\AppData\Local\Temp\npm-9488-bbf12b09\registry.npmjs.org\merge\-\merge-1.2.0.tgz`. I've just done `npm cache clean` but the error persists.

**Comment by kimblim** [comment]
Even if I disable the indentation check (by setting it to 0) I still get the following error:
```
/Users/kim/Sites/TDC-YS/frontend/node_modules/sass-lint/lib/rules/indentation.js:52
                  'line': node.content[i + 1].start.line,
                                             ^
TypeError: Cannot read property 'start' of undefined
    at processNode (/Users/kim/Sites/TDC-YS/frontend/node_modules/sass-lint/lib/rules/indentation.js:52:46)
    at processNode (/Users/kim/Sites/TDC-YS/frontend/node_modules/sass-lint/lib/rules/indentation.js:77:9)
    at processNode (/Users/kim/Sites/TDC-YS/frontend/node_modules/sass-lint/lib/rules/indentation.js:77:9)
    at processNode (/Users/kim/Sites/TDC-YS/frontend/node_modules/sass-lint/lib/rules/indentation.js:77:9)
    at processNode (/Users/kim/Sites/TDC-YS/frontend/node_modules/sass-lint/lib/rules/indentation.js:77:9)
    at Object.module.exports.detect (/Users/kim/Sites/TDC-YS/frontend/node_modules/sass-lint/lib/rules/indentation.js:81:5)
    at /Users/kim/Sites/TDC-YS/frontend/node_modules/sass-lint/index.js:44:27
    at Array.forEach (native)
    at Function.sassLint.lintText (/Users/kim/Sites/TDC-YS/frontend/node_modules/sass-lint/index.js:43:11)
    at DestroyableTransform._transform (/Users/kim/Sites/TDC-YS/frontend/node_modules/gulp-sass-lint/index.js:32:27)
```

I'm on El Capitan, node version 0.10.24.

Anyone who can point me in the right direction, even if I need to "hack" something for now.

**Comment by brokentone** [comment]
Having the same issue with tabs at a nest of 2:

Rules
```
indentation:
  - 1
  -
    size: 4
```

Erroring code:
```
.error-page {
	.content-area {
		right: 0;
	}
}
```

**Comment by Saturn2888** [comment]
I'm having the same issue even with spaces even if my SCSS is only this:

```scss
.something {
    font-weight: normal;
}
```

**Comment by DanPurdy** [comment]
@Saturn2888 are there any other files within the same directory? i've run that code through our tests and there's no problem with it. Could you upload the actual file?

**Comment by DanPurdy** [comment]
Some of these issues may have been rectified in #314 with regards to the indentation rule reporting errors when switched off in the config as for the others once the new version of the AST we use drops we'll be updating some of our rules and indentation is probably due for it.

**Comment by tnguyen14** [comment]
@DanPurdy any guess on when sass-lint will use that new gonzales version?

**Comment by DanPurdy** [comment]
@tnguyen14 we're still running through all the issues we've had reported here and gonzales is still seeing regular updates, once we're seeing all of our issues resolved we'll need to update some of our rules due to the breaking changes that have been made but that shouldn't take long. I can't really give an accurate time frame I'm afraid but it shouldn't be too long now.

**Comment by tnguyen14** [comment]
@DanPurdy ok great! thanks! I'm trying to implement sass-lint for our project as our .scss files are getting way complicated and we're having more people collaborating, so need to ensure some resemblance of standards.

Since we use tabs for all of our .scss files, this issue becomes a blocker for us as we cannot run the linter at all.

**Comment by DanPurdy** [comment]
Yeah we hear you,  definitely! 

All I can say is rest assured it will be a priority once it's in a state we're comfortable with. Tony the developer of gonzales-pe is doing great things right now and we're trying to be as helpful as possible to make sure nothing we know about gets missed.

**Comment by wouterkroes** [comment]
First of all, thanks @DanPurdy and others for all the great work so far! 
Hopefully this problem is solved soon ;)

**Comment by zenatreat** [comment]
bump - any updates on this?

**Comment by Snugug** [comment]
Still waiting on updates from our AST

**Comment by ebryn** [comment]
Any news?

**Comment by DanPurdy** [comment]
Not yet, you can check the status and issues of gonzales-pe [here](https://github.com/tonyganch/gonzales-pe)

**Comment by ebryn** [comment]
Is there a specific issue to follow?

**Comment by DanPurdy** [comment]
Unfortunately no, on last test it's actually fixed in the latest version of gonzales, the only problem is it breaks a few other bits and pieces that would end up breaking our support in other places so we're reluctant to move until it's stable.

**Comment by zenatreat** [comment]
ping, any updates?

**Comment by bgriffith** [comment]
This is fixed in the latest version of the AST we use but we can't upgrade yet as there's an issue blocks by it from installing on Windows. As soon as that's fixed we'll be pushing the update.

**Comment by zenatreat** [comment]
hallelujah!

**Comment by jjalonso** [comment]
:( Failing here still. Sorry.

**Comment by DanPurdy** [comment]
@jjalonso that's probably because we haven't released the update yet... :smile: Hopefully soon!

**Comment by jjalonso** [comment]
Go for it! Good work!

**Comment by chielkunkels** [comment]
Any eta on when this will be released? My team is always using tabs so I can't really introduce this until I can use it on files with tabs. Great job though!

**Comment by theneverstill** [comment]
@chielkunkels Ditto

**Comment by chielkunkels** [comment]
I've actually started using this now, but I'm not validating the whitespace just yet. Some checking is still better than nothing at all! :)

**Comment by DanPurdy** [comment]
We're at a bit of a crossroads at the moment, while running through our tests on all platforms we've found the latest version of the AST fixes a lot of our issues but also breaks a few things. Specifically Sass format on windows mainly. There's also some pretty annoying windows bugs left over so we're currently debating and looking into what our best course of action is. I hope to have 1.6 out soon though... I just can't say when.

**Comment by jjalonso** [comment]
This is working in any other old version? Can we downgrade ?

**Comment by DanPurdy** [comment]
@jjalonso @chielkunkels @zenatreat @davidgilbertson @wouterkroes @kimblim I believe the error that was being thrown was 'fixed' previously but with the release of sass-lint 1.6.0 we have tested all of your examples above and each of them is parsing correctly. Tabs on both LF and CRLF and spaces on LF and CRLF are all working. We recently fixed the CRLF issue ourselves in a temporary fork of the AST so if we find any extra issues there we can hopefully look to fix them.

As a warning though, the indentation rule itself may still be throwing some inconsistent lint warnings but we'll be looking to improve that now that 1.6.0 is FINALLY out of the door! If you spot any further issues here please let us know.

Thanks for your patience!
